[frontend] quiz dialog refactoring

sebastien-comeau opened the pull request:
### Description

List of proposed changes:

- use [LoadingButton ](button) when submitting the quiz
- refactoring logic and style of all Question components

### What to test for/How to test

### Additional Notes

This PR wwitch custom ToggleButton style to MUI default style. The current style is identical to a regular MUI Button and it's confusing. 

https://mui.com/material-ui/react-toggle-button/

Fbasham commented:
Mostly cosmetic things, feel free to tell me to GTFO:

My windows are wide open with sun coming in and the colour contrast on the button when it's selected seems really low.  I found it hard to tell it was selected.  The image below makes it look okay and it's clear what the answer is, but I'm wondering if increasing the colour contrast would be a better option, especially for those with low vision.

![image](https://github.com/DTS-STN/senior-journey/assets/48450599/df3fb512-939f-4e5e-9ab1-5d38fccad0ce)

---

Curious about the "next" and "previous" button placement.  On mobile, you have them side-by-side, but in the mockup, they're stacked:

![image](https://github.com/DTS-STN/senior-journey/assets/48450599/e34e3f5f-7237-4996-9373-3312b33030b4)


---

Also curious about the decision to have the previous button disabled on the first question when the mockup has it hidden:

![image](https://github.com/DTS-STN/senior-journey/assets/48450599/63d2487b-7887-4868-85e4-93a78f901cc1)

sebastien-comeau commented:
> Mostly cosmetic things, feel free to tell me to GTFO:
> 
> My windows are wide open with sun coming in and the colour contrast on the button when it's selected seems really low. I found it hard to tell it was selected. The image below makes it look okay and it's clear what the answer is, but I'm wondering if increasing the colour contrast would be a better option, especially for those with low vision.
> 

I've checked it it pass a11y, no issue.

![image](https://github.com/DTS-STN/senior-journey/assets/114004123/2edb7571-6906-4fbe-9a9a-650de057fa96)

> 
> Curious about the "next" and "previous" button placement. On mobile, you have them side-by-side, but in the mockup, they're stacked:
> 

Fixed.

> 
> Also curious about the decision to have the previous button disabled on the first question when the mockup has it hidden:
> 

I'm used to see and use pagination with previous disabled when on first page. Fixed.

Fbasham reviewed (approved):
Looks great.  Nice job.
